src/mlpack/core/tree/spill_tree/spill_tree.hpp

```diff
@@ -0,0 +1,432 @@
+  //! things up.
+  ElemType furthestDescendantDistance;
+  //! The minimum distance from the center to any edge of the bound.
+  ElemType minimumBoundDistance;
+  //! The dataset.  If we are the root of the tree, we own the dataset and must
+  //! delete it.
+  const MatType* dataset;
+  //! If true, we own the dataset and need to destroy it in the destructor.
+  bool localDataset;
+
+ public:
+  //! A single-tree traverser for hybrid spill trees; see
+  //! single_tree_traverser.hpp for implementation.
+  template<typename RuleType>
+  class SingleTreeTraverser;
+
+  //! A dual-tree traverser for hybrid spill trees; see dual_tree_traverser.hpp.
+  template<typename RuleType>
+  class DualTreeTraverser;
+
+  /**
+   * Construct this as the root node of a hybrid spill tree using the given
+   * dataset.  The dataset will not be modified during the building procedure
+   * (unlike BinarySpaceTree).
+   *
+   * @param data Dataset to create tree from.  This will be copied!
+   * @param tau Overlapping size.
+   * @param maxLeafSize Size of each leaf in the tree.
+   * @param rho Balance threshold.
+   */
+  SpillTree(const MatType& data,
+            const double tau = 0,
+            const size_t maxLeafSize = 20,
+            const double rho = 0.7);
+
+  /**
+   * Construct this as the root node of a hybrid spill tree using the given
+   * dataset.  This will take ownership of the data matrix; if you don't want
+   * this, consider using the constructor that takes a const reference to a
+   * dataset.
+   *
+   * @param data Dataset to create tree from.
+   * @param tau Overlapping size.
+   * @param maxLeafSize Size of each leaf in the tree.
+   * @param rho Balance threshold.
+   */
+  SpillTree(MatType&& data,
+            const double tau = 0,
+            const size_t maxLeafSize = 20,
+            const double rho = 0.7);
+
+  /**
+   * Construct this node as a child of the given parent, starting at column
+   * begin and using count points.  The ordering of that subset of points in the
+   * parent's data matrix will be modified!  This is used for recursive
+   * tree-building by the other constructors which don't specify point indices.
+   *
+   * @param parent Parent of this node.  Its dataset will be modified!
+   * @param points Vector of indexes of points to be included in this node.
+   * @param tau Overlapping size.
+   * @param maxLeafSize Size of each leaf in the tree.
+   * @param rho Balance threshold.
+   */
+  SpillTree(SpillTree* parent,
+            std::vector<size_t>& points,
+            const double tau = 0,
+            const size_t maxLeafSize = 20,
+            const double rho = 0.7);
+
+  /**
+   * Create a hybrid spill tree by copying the other tree.  Be careful!  This
+   * can take a long time and use a lot of memory.
+   *
+   * @param other tree to be replicated.
+   */
+  SpillTree(const SpillTree& other);
+
+  /**
+   * Move constructor for a SpillTree; possess all the members of the
+   * given tree.
```

Comment:
This looks like it's shorter than 80 characters, so I think you can compress it onto a single line.  I know, trivial comment :)